generator.ts

```diff
@@ -1,47 +1,61 @@
 import { promises as fsPromises } from 'fs';
 
-import _ from 'lodash';
 import yaml from 'js-yaml';
+import _ from 'lodash';
+import { OpenAPI, OpenAPIV3 } from 'openapi-types';
+import 'source-map-support/register';
 
 import { init } from './init';
 import {
   getResourceSchema,
   getResultSchema,
   getSetResultSchema,
-  getPostBodySchema,
-  getPatchBodySchema,
+  getRequestBodySchema,
 } from './schemas';
+import { Config, Resource } from './types';
 
 /**
  * Gets the prefix of the resource schema in components/schemas
  *
  * @param resourceName
  * @returns The prefix
  */
-const getResourceSchemaPrefix = (resourceName: string) => _.capitalize(resourceName);
+const getResourceSchemaPrefix = (resourceName: string): string => _.capitalize(resourceName);
 
 /**
- * Gets the method for a given endpoint. Example: `getById` -> `get`
+ * Gets the method for a given operation. Example: `getById` -> `get`
  *
- * @param endpoint
+ * @param operation
  * @returns The method
+ * @throws Error when operation is not in a valid format
  */
-const getEndpointMethod = (endpoint: string): string => endpoint.match(/^[a-z]+/)![0];
+const getOperationMethod = (operation: string): string => {
+  const operationRegex = /^[a-z]+/;
+  const match = operationRegex.exec(operation);
+  if (!match) {
+    throw Error('Invalid operation');
+  }
+  return match[0];
+};
 
 /**
- * Generates the value for `operationId`
+ * Generates the value for `operationId`. Example: `getById` -> `getPetById`
  *
- * @param endpoint
+ * @param operation
  * @param resourceName
  * @param resource - The resource object
  * @returns The value of operationId
  */
-const getOperationId = (endpoint: string, resourceName: string, resource: any): string => {
-  if (endpoint === 'get') {
-    return `${endpoint}${_.capitalize(resource.plural)}`;
+const getOperationId = (operation: string, resourceName: string, resource: Resource): string => {
+  if (operation === 'get') {
+    return `${operation}${_.capitalize(resource.plural)}`;
   }
-  const words = _.words(endpoint, /(^[a-z]+)|([A-Z][a-z]*)/g);
-  words.splice(1, 0, `${_.capitalize(resourceName)}`);
+  const words = _.words(operation, /(^[a-z]+)|([A-Z][a-z]*)/g);
+
+  // make sure resource name is in camel case and capitalize first letter
+  let resourcePart = _.camelCase(resourceName);
+  resourcePart = `${resourcePart.charAt(0).toUpperCase()}${resourcePart.slice(1)}`;
+  words.splice(1, 0, resourcePart);
   return _.join(words, '');
```

Comment:
This is what I meant (equivalent to line 55-59):

```js
words.splice(1, 0, resourceName);
return _.camelCase(_.join(words, '-'))
```

Reply:
Nice, this is a lot simpler.